java/ql/src/experimental/Security/CWE/CWE-200/SensitiveAndroidFileLeak.ql

```diff
@@ -0,0 +1,83 @@
+/**
+ * @name Leaking sensitive Android file
+ * @description Getting file intent from user input without path validation could leak arbitrary
+ *              Android configuration file and sensitive user data.
+ * @kind path-problem
+ * @id java/sensitive-android-file-leak
+ * @problem.severity warning
+ * @tags security
+ *       external/cwe/cwe-200
+ */
+
+import java
+import semmle.code.java.controlflow.Guards
+import AndroidFileIntentSink
+import AndroidFileIntentSource
+import DataFlow::PathGraph
+
+private class StartsWithSanitizer extends DataFlow::BarrierGuard {
+  StartsWithSanitizer() { this.(MethodAccess).getMethod().hasName("startsWith") }
+
+  override predicate checks(Expr e, boolean branch) {
+    e =
+      [
+        this.(MethodAccess).getQualifier(),
+        this.(MethodAccess).getQualifier().(MethodAccess).getQualifier()
+      ] and
+    branch = false
+  }
+}
+
+class AndroidFileLeakConfig extends TaintTracking::Configuration {
+  AndroidFileLeakConfig() { this = "AndroidFileLeakConfig" }
+
+  /**
+   * Holds if `src` is a read of some Intent-typed method argument guarded by a check like
+   * `requestCode == REQUEST_CODE__SELECT_CONTENT_FROM_APPS`, where `requestCode` is the first
```

Comment:
Clarify it isn't necessarily `REQUEST_CODE__SELECT_CONTENT_FROM_APPS` but rather any request code such that a call to `startActivityForResult(_, requestCode)` exists somewhere.

Reply:
Added an explanation of <code>REQUEST_CODE__SELECT_CONTENT_FROM_APPS</code>